src/main/java/am/ik/yavi/constraint/time/ChronoZonedDateTimeConstraintBase.java

```diff
@@ -0,0 +1,42 @@
+package am.ik.yavi.constraint.time;
+
+import am.ik.yavi.core.Constraint;
+import am.ik.yavi.core.ConstraintPredicate;
+
+import java.time.ZoneId;
+import java.time.chrono.ChronoZonedDateTime;
+import java.util.function.Predicate;
+
+import static am.ik.yavi.core.NullAs.VALID;
+import static am.ik.yavi.core.ViolationMessage.Default.*;
+
+/**
+ * This is the base class for constraints on ChronoZonedDateTime.
+ *
+ * @author Diego Krupitza
+ */
+public abstract class ChronoZonedDateTimeConstraintBase<T, V extends ChronoZonedDateTime<?>, C extends Constraint<T, V, C>>
+		extends ComparableTemporalConstraintBase<T, V, C> {
+
+	/**
+	 * Is the given ChronoZonedDateTime in the same zone as {@code zoneId}
+	 *
+	 * @param zoneId the zone our object has to be too
+	 */
+	public C zone(ZoneId zoneId) {
```

Comment:
What are the use cases that require this constraint?

Reply:
Maybe we want to validate that the given Object has the zone `Asia/Tokyo` before comparing it to other Objects. Since different zones in comparing time makes a huge difference.
Example:

```java
LocalDateTime localNow = LocalDateTime.now();
		
ZoneId tokyoZone = ZoneId.of("Asia/Tokyo");
ZoneId europeZone = ZoneId.of("Europe/Paris");

System.out.println(localNow.atZone(tokyoZone).compareTo(localNow.atZone(europeZone)));
System.out.println(localNow.atZone(tokyoZone).equals(localNow.atZone(europeZone)));
```

The output will be 
```
-1
false
```

Reply:
Let's remove this method. If you want to constrain `ZoneId` to `ZonedDateTime`, I think you should use `LocalDateTime` instead.

Reply:
Ok I will delete this method